sdk/core/Azure.Core/tests/SyncAsyncEventHandlerTests.cs

```diff
@@ -0,0 +1,700 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Pipeline;
+using Azure.Core.TestFramework;
+using NUnit.Framework;
+
+namespace Azure.Core.Tests
+{
+    public class SyncAsyncEventHandlerTests : ClientTestBase
```

Comment:
Need tests when event is raised from sync context

Reply:
Do you mean from a sync code path?  We're relying on the test framework's switch between sync/async methods (the fake `TestClient` is instrumented so `DoWorkAsync` is automagically changed to `DoWork`).  Or are you talking about parameterizing across other SynchronizationContexts?  I'm not opposed, but I haven't seen this anywhere else.